libtransmission/web.cc

```diff
@@ -431,6 +431,31 @@ class tr_web::Impl
         auto* task = static_cast<Task*>(vtask);
         TR_ASSERT(std::this_thread::get_id() == task->impl.curl_thread->get_id());
 
+        if (auto const range = task->range(); range)
+        {
+            // https://curl.se/libcurl/c/CURLINFO_RESPONSE_CODE.html
+            // "The stored value will be zero if no server response code has been received"
+            static auto constexpr NoResponseCode = 0L;
+            static auto constexpr PartialContentResponseCode = 206L;
+
+            // Test for webservers that don't support partial-content, see GH #4595
+            auto code = long{};
+            (void)curl_easy_getinfo(task->easy(), CURLINFO_RESPONSE_CODE, &code);
+            if (code != NoResponseCode && code != PartialContentResponseCode)
+            {
+                tr_logAddWarn(fmt::format(
+                    _("Couldn't fetch '{url}': expected HTTP response code {expected_code}, got {actual_code}"),
```

Comment:
Note, this does introduce a new string for translation, and since we're pretty close to release the chances are that this won't be translated in 4.0.0.

That's not great; but since it's better than not logging this error at all, IMO this should still go into 4.0.0